bpython/curtsiesfrontend/replpainter.py

```diff
@@ -161,6 +161,12 @@ def formatted_argspec(funcprops, arg_pos, columns, config):
 
 
 def formatted_docstring(docstring, columns, config):
+    if isinstance(docstring, bytes):
+        docstring = docstring.decode('utf8')
+    elif isinstance(docstring, str if py3 else unicode):
+        pass
+    else:
+        return []
```

Comment:
To answer your question, docstrings should always be unicode in python 3, and in Python 2 they should always be bytestrings. (since we're getting them from pydoc.getdoc, which does this normalization) If we got a unicode string somehow in Python 2 that would be ok, but I don't know how that would happen. If we got a bytestring in Python3, which shouldn't happen, we would try to decode. So this does cover all valid cases, but it covers some extra too.

Now that I see where docstring comes from (`pydoc.getdoc`) I agree that the else isn't necessary.

The correct thing to do here is to find out the encoding of the source file the docstring comes from, since it doesn't have to be utf8, or at least catch errors here so a bad docstring doesn't crash bpython.